### PR TITLE
Renamed some Vector methods to keep consistency

### PR DESCRIPTION
**Pull request by novialriptide**
This is in response to https://github.com/pygame/pygame/issues/2775

### CODE REVIEW

**Comment by Starbuck5** [comment]
Hey, thanks for this.

I see that you treated the old methods as aliases, which works.

However, I would like them to also emit deprecation warnings. Here's an example of a deprecation warning from c code: https://github.com/pygame/pygame/blob/main/src_c/mouse.c#L379-L381. And then you should be able to forward the PyObject args and kwargs (potentially) into the proper function.

**Comment by novialriptide** [comment]
@Starbuck5 It looks my pr is ready

**Review by ankith26 — COMMENTED** [review]
A mention of the deprecated functions in the docs might be a good thing to do :)

File: src_c/math.c [hunk 1]
```diff
@@ -3054,24 +3101,32 @@ static PyMethodDef vector3_methods[] = {
...
+    {"rotate_y_rad_ip", (PyCFunction)vector3_rotate_y_rad_ip, METH_O,
+     DOC_VECTOR3ROTATEYIPRAD},
     {"rotate_y_ip_rad", (PyCFunction)vector3_rotate_y_ip_rad, METH_O,
      DOC_VECTOR3ROTATEYIPRAD},
     {"rotate_z", (PyCFunction)vector3_rotate_z, METH_O, DOC_VECTOR3ROTATEZ},
     {"rotate_z_ip", (PyCFunction)vector3_rotate_z_ip, METH_O,
      DOC_VECTOR3ROTATEZIP},
     {"rotate_z_rad", (PyCFunction)vector3_rotate_z_rad, METH_O, DOC_VECTOR3ROTATEZRAD},
+    {"rotate_z_rad_ip", (PyCFunction)vector3_rotate_z_rad_ip, METH_O,
+     DOC_VECTOR3ROTATEZIPRAD},
```
Comment (ankith26) [inline src_c/math.c:3129]: the doc macro name on this line should be fixed. Also the same fix must be applied on other functions

**Review by ankith26 — COMMENTED** [review]
Ok left some more reviews. But overall this PR is looking great! Infact we were just discussing on discord whether this feature should make it to pygame 2.1.1, with Starbuck5 supporting this feature to make it in, at the earliest! I'm fine with this coming in any version really, so perhaps we need more people to review and weigh in their thoughts to decide which version this gets added

**Review by MyreMylar — APPROVED** [review]
These changes all look good to me, the new naming is more consistent with the current pattern in the math module. It is also a good work out for deprecation warnings which I expect us to occasionally need to exercise in the future of pygame. 

EDIT: I say get it in ASAP as the faster it is in the less time there is for people to use the old API.

👍

File: src_c/doc/math_doc.h [hunk 2]
```diff
@@ -21,7 +21,7 @@
 #define DOC_VECTOR2ROTATE "rotate(angle) -> Vector2\nrotates a vector by a given angle in degrees."
 #define DOC_VECTOR2ROTATERAD "rotate_rad(angle) -> Vector2\nrotates a vector by a given angle in radians."
 #define DOC_VECTOR2ROTATEIP "rotate_ip(angle) -> None\nrotates the vector by a given angle in degrees in place."
-#define DOC_VECTOR2ROTATEIPRAD "rotate_ip_rad(angle) -> None\nrotates the vector by a given angle in radians in place."
+#define DOC_VECTOR2ROTATEIPRAD "rotate_rad_ip(angle) -> None\nrotates the vector by a given angle in radians in place."
```
Comment (ankith26) [inline src_c/doc/math_doc.h:24]: The doc macro should read `DOC_VECTOR2ROTATERADIP` here :)

PS: These doc headers are autogenerated by running `python3 setup.py docs`, you don't have to manually change these

File: src_c/math.c [hunk 3]
```diff
@@ -2059,6 +2059,15 @@ vector2_rotate_ip_rad(pgVector *self, PyObject *angleObject)
     Py_RETURN_NONE;
 }
 
+static PyObject *
+vector2_rotate_ip_rad(pgVector *self, PyObject *angleObject)
+{
+    if (PyErr_WarnEx(PyExc_DeprecationWarning, "vector2_rotate_rad_ip() now has all the functionality of vector2_rotate_ip_rad(), so vector2_rotate_ip_rad() will be deprecated in pygame 2.2", 1) == -1) {
```
Comment (ankith26) [inline src_c/math.c:2065]: the version number here needs updating. And similar changes needed in other functions

**Review by ankith26 — APPROVED** [review]
Looks good to go now! Thanks for the PR! 🎉 👍

**Comment by Starbuck5** [comment]
Alright, this is really close to ready.

The old ones could be untyped, but we're leaving it documented as deprecated, so it makes sense to leave the types as well, I suppose.

The final thing is the the _doc.h files are auto generated on the `python setup.py docs`. It looks like you edited these ones manually, since it changes when the docs are regenerated. If you generate them automatically the old and new method docstrings are both there, so I suppose the C code should use the correct docstring for each.

Also maybe whoever merges this should do a squash and merge, since there are lots of small commits?

**Comment by novialriptide** [comment]
I squashed all the commits into one and regenerated the doc files.

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

**Review by ankith26 — APPROVED** [review]
:+1: Thanks for the PR!

**Pull request merged**